pipelines/shared/nextflow/color_workflow.nf

```diff
@@ -0,0 +1,202 @@
+        --cluster-num-map $cluster_num_map --cluster-color-map cluster_colors.txt --color-file $projectDir/../shared/perl/colors.tab
+    """
+}
+
+process get_ssn_id_info {
+    input:
+        path ssn_file
+    output:
+        path 'edgelist.txt', emit: 'edgelist'
+        path 'index_seqid_map.txt', emit: 'index_seqid_map'
+        path 'id_index_map.txt', emit: 'id_index_map'
+        path 'seqid_source_map.txt', emit: 'seqid_source_map'
+    """
+    perl $projectDir/../shared/perl/ssn_to_id_list.pl --ssn $ssn_file --edgelist edgelist.txt --index-seqid index_seqid_map.txt \
+        --id-index id_index_map.txt --seqid-source-map seqid_source_map.txt
+    """
+}
+
+process unzip_input {
+    input:
+        path ssn_zipped
+    output:
+        path "ssn____local.xgmml"
+    """
+    perl $projectDir/../shared/perl/unzip_xgmml_file.pl --in $ssn_zipped --out ssn____local.xgmml
+    """
+}
+
+// This is necessary to avoid Docker mounting the same directory twice; this occurs
+// because we need to mount the root of the project so that we can get access to the
+// lib directory. When using a SSN that is inside the project directory then the project
+// directory is also mounted.
+process copy_input {
+    input:
+        path ssn_file
+    output:
+        path "ssn____local.xgmml"
+    """
+    cp $ssn_file ssn____local.xgmml
+    """
+}
+
+process get_annotated_mapping_tables {
+    publishDir params.final_output_dir, mode: 'copy'
+    input:
+        path cluster_id_map
+        path seqid_source_map
+        path cluster_color_map
+    output:
+        path 'mapping_table.txt', emit: 'mapping_table'
+        path 'swissprot_clusters_desc.txt', emit: 'swissprot_table'
+    """
+    perl $projectDir/../shared/perl/annotate_mapping_table.pl --seqid-source-map $seqid_source_map --cluster-map $cluster_id_map \
+        --cluster-color-map $cluster_color_map --mapping-table mapping_table.txt --swissprot-table swissprot_clusters_desc.txt \
+        --config ${params.efi_config} --db-name ${params.efi_db}
+    """
+}
+
+process get_conv_ratio_table {
+    publishDir params.final_output_dir, mode: 'copy'
+    input:
+        path edgelist
+        path index_seqid_map
+        path cluster_id_map
+        path seqid_source_map
+    output:
+        path 'conv_ratio.txt', emit: 'conv_ratio'
+    """
+    perl $projectDir/../shared/perl/compute_conv_ratio.pl --cluster-map $cluster_id_map --index-seqid-map $index_seqid_map \
+        --edgelist $edgelist --seqid-source-map $seqid_source_map --conv-ratio conv_ratio.txt
+    """
+}
+
+process get_cluster_stats {
+    publishDir params.final_output_dir, mode: 'copy'
+    input:
+        path cluster_id_map
+        path seqid_source_map
+        path singletons
+    output:
+        path 'stats.txt', emit: 'stats'
+    """
+    perl $projectDir/../shared/perl/compute_stats.pl --cluster-map $cluster_id_map --seqid-source-map $seqid_source_map \
+        --singletons $singletons --stats stats.txt
```

Comment:
`conv_ratio` and `compute_stats` both exist as python functions. I think they both accept parquet files so they obviously wouldn't work for this but in the future it would be good to define those computations in terms of nodes and edges exactly once and then only write code to adapt the input type to work with those functions. The calculations could be Perl or Python, the important thing would be the ability to read diverse file types.